scripts/generate_road_colors.py

```diff
@@ -0,0 +1,227 @@
+from colormath.color_conversions import convert_color
+from colormath.color_objects import LabColor, LCHabColor, SpectralColor, sRGBColor, \
+XYZColor, LCHuvColor, IPTColor
+from colormath.color_diff import delta_e_cie2000
+import numpy
+
+def int_lch_to_string(lch):
+	lch = lch.get_value_tuple()
+	l = "{0:.1f}".format(lch[0])
+	c = "{0:.1f}".format(lch[1])
+	h = "{0:.1f}".format(lch[2])
+	if l[-1] == '0':
+ 	  l = "{0:.0f}".format(lch[0])
+	if c[-1] == '0':
+ 	  c = "{0:.0f}".format(lch[1])
+	if h[-1] == '0':
+ 	  h = "{0:.0f}".format(lch[2])
+
+	return 'lch(' + l + ', ' + c + ', ' + h + ')' 
+
+
+class Color:
+	def __init__(self, name, hex):
+		rgb = sRGBColor.new_from_rgb_hex(hex)
+		self.lch = convert_color(rgb, LCHabColor)
+		if name != None:
+			if 'low-zoom' in name:
+				self.lch.lch_c = 50
+			if 'casing' in name:
+				self.lch.lch_l = 50
+				self.lch.lch_c = 80
+		self.intended_lch_string = int_lch_to_string(self.lch)
+		self.hex = str(convert_color(self.lch, sRGBColor).get_rgb_hex())
+		assert(len(self.hex) == 7)
+
+	@classmethod
+	def lch(self, l, c, h):
+		lch = LCHabColor(l, c, h)
+		rgb = convert_color(lch, sRGBColor)
+		#print(lch.get_value_tuple())
+		#print(rgb.get_upscaled_value_tuple())
+		hex = str(rgb.get_rgb_hex())
+		if (len(hex) != 7):
+			print("this lch value is outside rgb range")
+			exit
+
+		effective_lch = convert_color(rgb, LCHabColor)
+		effective_lch_tuple = effective_lch.get_value_tuple()
+		lch_tuple = lch.get_value_tuple()
+
+		returned = Color(None, hex)
+
+		color1_lab = convert_color(effective_lch, LabColor);
+		color2_lab = convert_color(lch, LabColor);
+		delta_e = delta_e_cie2000(color1_lab, color2_lab);
+
+		if delta_e > 2.5:
+			print("high conversion error on moving to rgb")
+
+		delta = tuple(numpy.subtract(effective_lch_tuple, lch_tuple))
+		lch_delta = ' lch(' + str("{0:.1f}".format(delta[0])) + ', ' + str("{0:.1f}".format(delta[1])) + ', ' + str("{0:.1f}".format(delta[2])) + ')' 
+
+		returned.intended_lch_string = int_lch_to_string(lch) + ' Conversion error on moving from lch to rgb: ' + str("{0:.1f}".format(delta_e)) + lch_delta
+
+		return returned
+
+	def get_as_hex(self):
+		return self.hex
+
+	def get_intended_lch(self):
+		return self.intended_lch_string
+
+min_h = 10
+max_h = 106
+delta_h = (max_h - min_h) / 3
+motorway_hue = (min_h) % 360
+trunk_hue = (min_h + delta_h) % 360
+primary_hue = (min_h + delta_h*2) % 360
+secondary_hue = (min_h + delta_h*3) % 360
+
+colors = {}
+
+min_l = 65
+max_l = 80
+delta_l = (max_l - min_l)/3
+
+c = 30
+
+i = 0
+colors['motorway_shield'] = Color.lch(min_l+delta_l*i, c, motorway_hue)
+i = 1
+colors['trunk_shield'] = Color.lch(min_l+delta_l*i, c, trunk_hue)
+i = 2
+colors['primary_shield'] = Color.lch(min_l+delta_l*i, c, primary_hue)
+i = 3
+colors['secondary_shield'] = Color.lch(min_l+delta_l*i, c, secondary_hue)
+
+colors['tertiary_shield'] = Color.lch(min_l+delta_l*i, 0, 0)
+
+
+min_l = 70
```

Comment:
The section that calculates fill colors occurs twice (lines 101-114 and lines 116-132).